File: EX2/EX2/Material/Pregunta1-Parcial/TransitSoft/TransitSoftNegocio/CapturaBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;
using TransitSoftNegocio.ServiciosWS;

namespace TransitSoftNegocio
{
    public class CapturaBO
    {
        private ServicioCapturaClient capturaClientSOAP;

        public CapturaBO()
        {
            this.capturaClientSOAP = new ServicioCapturaClient();
        }

        BindingList<captura> ObtenerCapturasConExcesoDeVelocidad()
        {
            return new BindingList<captura>(this.capturaClientSOAP.obtenerCapturasConExcesoDeVelocidad());
        }
        bool Actualizar(captura modelo)
        {
            return this.capturaClientSOAP.actualizarCaptura(modelo);
        }
        public BindingList<captura> LeerTodos()
        {
            return new BindingList<captura>(this.capturaClientSOAP.listarTodasCapturas());
        }
        public int Insertar(captura capturaDTO)
        {
            return this.capturaClientSOAP.insertarCaptura(capturaDTO);
        }
        public int Modificar(captura capturaDTO)
        {
            return this.capturaClientSOAP.modificarCaptura(capturaDTO);
        }
        public int Eliminar(int id)
        {
            return this.capturaClientSOAP.eliminarCaptura(id);
        }
        public captura ObtenerPorId(int id)
        {
            return this.capturaClientSOAP.obtenerCapturaPorId(id);
        }
    }
}

File: EX2/EX2/Material/TransitSoft_ConexionMultiple-SOAP/TransitSoftNegocio/src/main/java/pe/edu/pucp/transitsoft/bo/CapturaBO.java
package pe.edu.pucp.transitsoft.bo;

import java.util.Date;
import java.util.List;
import pe.edu.pucp.transitsoft.modelo.Camara;
import pe.edu.pucp.transitsoft.modelo.Captura;
import pe.edu.pucp.transitsoft.modelo.EstadoCaptura;
import pe.edu.pucp.transitsoft.modelo.Vehiculo;

public interface CapturaBO {
    List<Captura> obtenerCapturasConExcesoDeVelocidad();
    boolean actualizar(int id, String placa, double velocidad, Date fechaCaptura, Camara camara,
            EstadoCaptura estado, Vehiculo vehiculo);
    public List<Captura> leerTodos();
    public Integer insertar(String placa, double velocidad, Date fechaCaptura, Camara camara,
            EstadoCaptura estado, Vehiculo vehiculo);
    public Integer modificar(int id, String placa, double velocidad, Date fechaCaptura, Camara camara,
            EstadoCaptura estado, Vehiculo vehiculo);
    public Integer eliminar(int id);
    public Captura obtenerPorId(int id);
}
